packages/vite/src/node/config.ts

```diff
@@ -427,6 +427,14 @@ export async function resolveConfig(
     }
   }
 
+  if (process.env.VITE_TEST_WITHOUT_PLUGIN_COMMONJS) {
+    config = mergeConfig(config, {
+      build: { commonjsOptions: { include: [] } },
```

Comment:
With `mergeConfig`, arrays are merged instead of replaced, so existing `build.commonjsOptions.include` would be left as it and commonjs plugin would still be included. Maybe we should hard set to an empty array after the merge?

Reply:
Given that this is an internal test env, I think we can keep it as a simple merge. If there are playgrounds specifying a custom `commonjsOptions.include`, they are probably meant to test the Rollup plugin commonjs specifically so better we respect it. But there aren't any tests like this AFAICS.

Reply:
If we want to remove plugin commonjs in the future, we should probably do a hard set regardless though, since ideally we want a nice transition for users when we switch internally. But I'm not hard on this since we're not running this on CI, so it's your call :)

That also reminds me that we should probably define `process.env.VITE_TEST_WITHOUT_PLUGIN_COMMONJS: false` in the rollup config so we treeshake it out in the final build.

Reply:
Ok, I modified it to a hard set. About removing it from the bundle, I think given that it only adds a few lines we could leave it as a way to test outside of Vite core.